src/util/address.rs

```diff
@@ -672,6 +672,13 @@ impl Address {
 
     /// Constructs an [`Address`] from an output script (`scriptPubkey`).
     pub fn from_script(script: &script::Script, network: Network) -> Option<Address> {
```

Comment:
It might not be to be done in this PR, but if we have multiple ways of "not turning script into an address", maybe we should make it `enum Error { ... list all the ways things go wrong ... }`

Reply:
Agreed. `Address::from_script()` and a few other functions should return `Result<_, Error>` but this is for another PR.

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/issues/1022 just to not forget.